packages/react-components/src/lib/context/NetworkProvider.tsx

```diff
@@ -0,0 +1,35 @@
+import { NetworkContext } from './NetworkContext';
+import { PropsWithChildren, useState } from 'react';
+import type { NetworkConfig } from './NetworkContext';
+
+type Props = { defaultNetworkConfig: NetworkConfig };
+
+const storageKey = 'agoric-saved-network-config';
+
+const readFromStorage = () => {
+  const data = localStorage.getItem(storageKey);
+  return data && JSON.parse(data);
+};
+
+const writeToStorage = (networkConfig: NetworkConfig) => {
+  const data = JSON.stringify(networkConfig);
+  localStorage.setItem(storageKey, data);
+};
+
+export const NetworkProvider = ({
+  defaultNetworkConfig,
+  children,
+}: PropsWithChildren<Props>) => {
+  const [networkConfig] = useState(readFromStorage() ?? defaultNetworkConfig);
+
+  const setNetworkConfig = (newConfig: NetworkConfig) => {
+    writeToStorage(newConfig);
+    window.location.reload();
```

Comment:
Is this needed? It looks like AgoricProvider is dependent on NetworkProvider, so the change should bubble down.

Reply:
Yes the change would bubble down, but imo basically all of the state coming from the wallet and chain would need to be cleaned up anyway. It's tricky to do that in `AgoricProviderLite` because the notifiers could only be cleaned up on the next iteration when the loop is broken, so they'll keep polling the old network until an update happens. And any additional state the app is watching ([example](https://github.com/Agoric/dapp-offer-up/blob/89dc44ec3701765a28ecb220e3e4671a53ef974b/ui/src/providers/Contract.tsx#L37)) would have to be cleaned up, so it's harder on the developer. This feels ugly but it's simple and safe, maybe there's some easy ways to get around those issues though.

Reply:
> the notifiers could only be cleaned up on the next iteration when the loop is broken

I might be missing something wrt this comment, but here's an approach i took to clean out state when the network config changed: https://github.com/0xpatrickdev/agoric-vault-collateralization-notifier/blob/d6729c6f43c2042ae783320462818c3e3275999f/web/src/contexts/chain.jsx#L34-L47